plugins/shadow-block-converter/src/shadow_block_converter.ts

```diff
@@ -32,9 +32,10 @@ export class BlockShadowStateChange extends Blockly.Events.BlockBase {
   /* eslint-enable @typescript-eslint/naming-convention */
 
   /**
-   * The index of the connection in the parent block's list of connections.
+   * The index of the connection in the parent block's list of connections. If
+   * null, then the nextConnection will be used instead.
    */
-  connectionIndex?: number;
+  inputIndexInParent?: number | null;
```

Comment:
Since requiring the constructor arguments to be defined, there's no need for this property to allow undefined values. (But yes they meant different things, null means that the connection is not associated with an input, i.e. it's the nextConnection.)